keyboards/snes_macropad/snes_macropad.c

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 John Barbero Unenge (@jbarberu)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+
+// oled keylog rendering has been kindly borrowed from crkbd <3
+
+char     key_name = ' ';
+uint16_t last_keycode;
+uint8_t  last_row;
+uint8_t  last_col;
+
+static const char PROGMEM code_to_name[60] = {' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'R', 'E', 'B', 'T', '_', '-', '=', '[', ']', '\\', '#', ';', '\'', '`', ',', '.', '/', ' ', ' ', ' '};
+
+static void set_keylog(uint16_t keycode, keyrecord_t *record) {
+    key_name     = ' ';
+    last_keycode = keycode;
+    if (IS_QK_MOD_TAP(keycode)) {
+        if (record->tap.count) {
+          keycode = QK_MOD_TAP_GET_TAP_KEYCODE(keycode);
+        } else {
+          keycode = 0xE0 + biton(QK_MOD_TAP_GET_MODS(keycode) & 0xF) + biton(QK_MOD_TAP_GET_MODS(keycode) & 0x10);
+        }
+    } else if (IS_QK_LAYER_TAP(keycode) && record->tap.count) {
+        keycode = QK_LAYER_TAP_GET_TAP_KEYCODE(keycode);
+    } else if (IS_QK_MODS(keycode)) {
+        keycode = QK_MODS_GET_BASIC_KEYCODE(keycode);
+    } else if (IS_QK_ONE_SHOT_MOD(keycode)) {
+        keycode = 0xE0 + biton(QK_ONE_SHOT_MOD_GET_MODS(keycode) & 0xF) + biton(QK_ONE_SHOT_MOD_GET_MODS(keycode) & 0x10);
+    }
+    if (keycode > ARRAY_SIZE(code_to_name)) {
+        return;
+    }
+
+    // update keylog
+    key_name = pgm_read_byte(&code_to_name[keycode]);
+    last_row = record->event.key.row;
+    last_col = record->event.key.col;
+}
+
+static const char *depad_str(const char *depad_str, char depad_char) {
+    while (*depad_str == depad_char) {
+        ++depad_str;
+    }
+    return depad_str;
+}
+
+static void oled_render_keylog(void) {
+    const char *last_row_str = get_u8_str(last_row, ' ');
+    oled_write(depad_str(last_row_str, ' '), false);
```

Comment:
Since this is only 0-10 (less)... you could just call: 
```suggestion
    oled_write('0' + last_row, false);
```

That saves a bunch of code and calculations. And greatly simplifies the code

Reply:
Had to use oled_write_char instead of oled_write, but otherwise worked like a charm!